src/renderer/object/texture_region.rs

```diff
@@ -0,0 +1,36 @@
+#[derive(Clone, Copy, Debug, Default)]
+pub struct TextureRegion {
+    pub x: f32,
+    pub y: f32,
+    pub w: f32,
```

Comment:
Call w and h something else, because it is a scale not the actual width and height.

Reply:
offset_x, offset_y, scale_x, scale_y

Not too long for being verbose, at least leans toward the spec.